src/gluonts/model/evaluation.py

```diff
@@ -130,16 +130,26 @@ def evaluate_forecasts_raw(
 
     index_data = []
 
-    for input_, label, forecast in tqdm(
-        zip(test_data.input, test_data.label, forecasts)
+    input_batches = batcher(test_data.input, batch_size=batch_size)
+    label_batches = batcher(test_data.label, batch_size=batch_size)
+    forecast_batches = batcher(forecasts, batch_size=batch_size)
+
+    pbar = tqdm()
```

Comment:
Can we also set the total length here?

Reply:
We could, but depending on what dataset type one is using, this may end up iterating the dataset to get its length. I don’t think the current code does it, so maybe let’s do it in a separate change.

Reply:
I guess you could try to call `len` on it as long as it doesn't consume it?

Reply:
Yes, that's the idea. However this may still take time (say the dataset is huge and `len` needs to iterate it), which I would like to avoid at least in this PR